### PR TITLE
[SPARK-29680][SQL][Followup] Replace qualifiedName with multipartIdentifier

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

Replace qualifiedName with multipartIdentifier in parser rules of DDL commands.

### Why are the changes needed?

There are identifiers in some DDL rules we use `qualifiedName`. We should use `multipartIdentifier` because it can capture wrong identifiers such as `test-table`, `test-col`.

### Does this PR introduce any user-facing change?

Yes. Wrong identifiers such as test-table, will be captured now after this change.

### How was this patch tested?

Unit tests.

### CODE REVIEW

**Comment by viirya** [comment]
actually I am not sure if this should keep as followup or should create another JIRA ticket. @cloud-fan

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 [hunk 1]
```diff
@@ -136,20 +136,22 @@ statement
     | ALTER TABLE multipartIdentifier
         ADD (COLUMN | COLUMNS)
         '(' columns=qualifiedColTypeWithPositionList ')'               #addTableColumns
+    | ALTER TABLE table=multipartIdentifier
+        RENAME COLUMN
+        from=multipartIdentifier TO to=multipartIdentifier             #renameTableColumn
```
Comment (cloud-fan) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: `to` should be a single `identifer`. We can only rename a column or a field, but not a field chain.

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 [hunk 2]
```diff
@@ -802,7 +808,7 @@ intervalUnit
     ;
 
 colPosition
-    : FIRST | AFTER qualifiedName
```
Comment (cloud-fan) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: can we remove the definition of `qualifiedName` if it's used nowhere?

Comment (viirya) [reply sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: in current change i just replaced qualifiedName for DDL commands.

there are still few places using qualifiedName. Need to address them see if we can replace them too.

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 (hunk 1 above)
Comment (viirya) [reply sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: oh, then it should be `errorCapturingIdentifier`.

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 (hunk 2 above)
Comment (viirya) [reply sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: Can a function be named as test-func without quote?

There are rules `CRATE (OR REPLACE)? TEMPORARY? FUNCTION...`  and `DROP TEMPORARY? FUNCTION` that use `qualifiedName`.

Comment (cloud-fan) [reply sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: `qualifiedName` can't match `a-b`. I take another look, `qualifiedName` is still needed in places where `-` is a legal following, e.g. `select a - b`, we must use `qualifiedName` to match column name.

For function name, we can use `multipartIdentifier`

Comment (viirya) [reply sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: If we use `multipartIdentifier` in CREATE FUNCTION, meaning function name is `multipartIdentifier`.  Will it be a problem for parsing `SELECT a-test_func(...) ...`?

In this case, a-test will be parsed by `multipartIdentifier` and an exception will be thrown to ask adding quote for it.

Comment (cloud-fan) [reply sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: We can only use `multipartIdentifier` when define the function, e.g. `CREATE FUNCTION ...`, not when calling the function.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ErrorParserSuite.scala [hunk 3]
```diff
@@ -106,6 +127,14 @@ class ErrorParserSuite extends AnalysisTest {
         |LOCATION '/user/external/page_view'
         |TBLPROPERTIES ('p1'='v1', 'p2'='v2')
         |AS SELECT * FROM src""".stripMargin, 2, 36, 37, msg + " page-view")
+    intercept(
+      """
+        |CREATE TABLE IF NOT EXISTS tab
+        |USING test-provider
+        |COMMENT 'This is the staging page view table'
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ErrorParserSuite.scala]: nit: we don't have to specify COMMENT, LOCATION, etc to test provider name.

**Review by cloud-fan — APPROVED** [review]

**Comment by viirya** [comment]
Thanks. Merging to master.

**Pull request closed without merging**